RocketChat/Rocket.Chat.ReactNative
# [IMPROVE] Apply skeleton for message images and on workspace avatar (#4810)

## Proposed changes

The goal of this PR is to implement the improvement mentioned in issue #4810, namely to display a skeleton placeholder when various images in the app are loading.

The approach I followed is roughly the one described [in the issue](https://github.com/RocketChat/Rocket.Chat.ReactNative/issues/4810#issuecomment-1463993988).

This PR is reviewed commit by commit.

## Issue(s)	

#4810

## How to test or reproduce

1. Clone [this branch made for QA](https://github.com/floels/Rocket.Chat.ReactNative/tree/FOR-QA/IMPROVE/4810-apply-skeleton-images) (sub-branch of this merge request's branch). I tweaked the `handleLoadEnd` method of the new `<ImageSkeletonPlaceholder/>` component to make placeholder skeletons visible for three seconds after the image has loaded. I also had to point to an arbitrary image to check the behavior for the server avatar of `open.rocket.chat` because it didn't show in my local environment, I'm not sure if this is a general issue or not.

2. Move to the cloned branch. Build and launch the app.

3. Tap `Join our open workspace`. Notice the placeholder skeleton appearing for three seconds before server avatar is loaded.

4. Log in to the open workspace and join public channel `test_issue_GH_4810`.

5. Send a message with an image attached. Notice the placeholder skeleton appearing for three seconds before the image is loaded.

6. Send another message in Markdown, for instance:

```markdown
![My image](https://i.imgur.com/KOdB2ed.png)
```

Notice the placeholder skeleton appearing for three seconds before the image is loaded.

NB: if you want to test the behavior on image load failure, comment/uncomment as indicated [here](https://github.com/floels/Rocket.Chat.ReactNative/blob/606ffbfb45a3f7119733b32c75c2f06707273616/app/containers/ImageWithSkeleton/index.tsx#L31-L32) and repeat the actions above. You should see the loading error message:

![Capture d’écran 2023-03-16 à 08 48 38](https://user-images.githubusercontent.com/79283128/225549569-c5f1fb0a-7ec7-4d1d-8fac-abed1f04aa90.png)

## Screenshots (iOS)

_NB: in the `After` GIFs, the image loading time was artificially set to three seconds, otherwise the effect is not visible._

### Server avatar

| Before | After |
| ------ | ----- |
| ![Simulator Screen Recording - iPhone 13 - 2023-03-11 at 20 50 39](https://user-images.githubusercontent.com/79283128/224508781-90daf71c-5b47-4a34-88d0-ab7a3982c63b.gif) | ![Simulator Screen Recording - iPhone 13 - 2023-03-11 at 20 41 47](https://user-images.githubusercontent.com/79283128/224508410-96d7ffe2-eb2b-4296-942a-6050c1487c78.gif) |

### Image in Markdown message

| Before | After |
| ------ | ----- |
| ![Simulator Screen Recording - iPhone 13 - 2023-03-11 at 20 47 23](https://user-images.githubusercontent.com/79283128/224508697-23e5f49a-afde-4db2-988e-023b34f9a79d.gif) | ![Simulator Screen Recording - iPhone 13 - 2023-03-11 at 20 39 53](https://user-images.githubusercontent.com/79283128/224508355-d1d25bc1-4d91-4f73-a3ee-7d314358dd74.gif) |

### Image in regular message

| Before | After |
| ------ | ----- |
| ![Simulator Screen Recording - iPhone 13 - 2023-03-11 at 20 46 05](https://user-images.githubusercontent.com/79283128/224508586-0773667c-fa7f-4bd9-802e-74ea6ef18f60.gif) | ![Simulator Screen Recording - iPhone 13 - 2023-03-11 at 20 37 13](https://user-images.githubusercontent.com/79283128/224508278-4bfe2fc2-eff1-40d3-a8cf-7f9ecf49966a.gif) |

## Types of changes




## Checklist


[NA] I have added necessary documentation (if applicable)
[NA] Any dependent changes have been merged and published in downstream modules
